### PR TITLE
Ports the Bot Control PDA program to Modular Computers.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
Hey, remember when we were going to make modular computers replace PDAs?
Yeah me neither.
## About The Pull Request

Ports the PDA App used by roboticists and heads of staff to control simplemob robots and MULEbots from the PDA over to modular computers. It's a 12 point program, so on par with the most expensive programs on modular computers, but with it downloaded you can turn any modular computer into a full control UI for managing to sending out simple mob robots. Useful for heads of staff to have on the modular computers on the bridge, as well as roboticists who invest in tablets.

Has the full bredth of features from the PDA Version, all working to the same level of accuracy as before. If the user has an id card slot component in their computer, they can also grant their ID's access to the robot when summoning it, as with the PDA.

Current version of the UI (4/2/20):
image

## Why It's Good For The Game

This project definitely fell by the wayside when, considering that there's still so few PDA apps, and PDA functions are all exceptionally more fleshed out. This might help get the ball rolling so that modular computers can start getting some of the existing PDA features as well.

## Changelog
:cl:
add: A new app is available for modular computers, the Bot Remote Controller, similar to the PDA Bot Control function.
/:cl:

### CODE REVIEW

**Comment by cacogen** [comment]
these are always so pleasing

File: tgui/packages/tgui/interfaces/NtosRoboControl.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                  </Box>
+                  <Box>
+                    Location: {robot.locat}
+                  </Box>
+                  <Box>
+                    Status: {robot.mode}
+                  </Box>
+                  {robot.mule_check === 1 &&(
+                    <Box>
+                      Loaded Cargo: {mules.load ? (mules.load) : "N/A"}
```
Comment (bobbahbrown) [inline tgui/packages/tgui/interfaces/NtosRoboControl.js:49]:
```suggestion
                      Loaded Cargo: {mules.load || "N/A"}
```

File: tgui/packages/tgui/interfaces/NtosRoboControl.js [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            <Grid.Column>
+              <Box>
+                {robot.mule_check === 1 &&(
+                  mules.map(mulebot => (
+                    (robot.bot_ref === mulebot.mule_ref) &&(
+                      <Box
+                        key={mulebot.mule_ref}
+                        textAlign="center">
+                        <Button
+                          tooltip="Stop Moving."
```
Comment (bobbahbrown) [inline tgui/packages/tgui/interfaces/NtosRoboControl.js:64]: for the tooltips you have mixed ending with a period and not ending with a period you should pick one and be consistent

File: tgui/packages/tgui/interfaces/NtosRoboControl.js [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                          content="Toggle Delivery Report"
+                          checked={reportDelivery}
+                          onClick={() => act('report', {
+                            robot: mulebot.mule_ref,
+                          })} />
+                      </Box>
+                    )
+                  ))
+                )}
+                <Box m={1} />
```
Comment (bobbahbrown) [inline tgui/packages/tgui/interfaces/NtosRoboControl.js:123]:
```suggestion
                <Box />
```

File: tgui/packages/tgui/interfaces/NtosRoboControl.js [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+                  {robot.mule_check === 1 &&(
+                    <Box>
+                      Loaded Cargo: {mules.load ? (mules.load) : "N/A"}
+                    </Box>
+                  )}
+                </Section>
+              </Box>
+            </Grid.Column>
+            <Grid.Column>
+              <Box>
```
Comment (bobbahbrown) [inline tgui/packages/tgui/interfaces/NtosRoboControl.js:56]:
```suggestion
              <Box m={1}>
```

Comment (bobbahbrown) [reply tgui/packages/tgui/interfaces/NtosRoboControl.js:56]: assuming you meant to put the one i suggested remmoving below up here

File: code/modules/modular_computers/file_system/programs/robocontrol.dm [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+		if("unload")
+			Bot.bot_control(action, current_user, current_access, TRUE)
+		if("autoret")
+			Bot.bot_control(action, current_user, current_access, TRUE)
+		if("autopick")
+			Bot.bot_control(action, current_user, current_access, TRUE)
+		if("report")
+			Bot.bot_control(action, current_user, current_access, TRUE)
+		if("ejectpai")
+			Bot.bot_control(action, current_user, current_access, TRUE)
```
[the remark this answers is not in the document]
Comment (bobbahbrown) [reply code/modules/modular_computers/file_system/programs/robocontrol.dm:119]: double also idk if you can do ``if (action in <thing>`` in the switch maybe double check that that's possible

if not you could just put those two outside the switch then switch the actual special cases

[the remark this answers is not in the document]
Comment (ArcaneMusic) [reply code/modules/modular_computers/file_system/programs/robocontrol.dm:119]: Apparently not possible, workshopping this part right now.

File: tgui/packages/tgui/interfaces/NtosRoboControl.js (hunk 2 above)
Comment (ArcaneMusic) [reply tgui/packages/tgui/interfaces/NtosRoboControl.js:64]: Tooltips all have periods, button content all don't have periods.

**Comment by GunHog** [comment]
More and more I wish the AI had an internal modular computer...this thing is so far above the Status page and normal borg console, it is CRAZY.

File: tgui/packages/tgui/interfaces/NtosRoboControl.js (hunk 2 above)
Comment (bobbahbrown) [reply tgui/packages/tgui/interfaces/NtosRoboControl.js:64]: oops didnt catch that >w<

File: tgui/packages/tgui/interfaces/NtosRoboControl.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+          onClick={() => act('ejectcard')}
+        />
+      )}>
+      <Box>
+        Id Card: {data.id_owner}
+      </Box>
+      <Box>
+        Bots detected in range: {data.botcount}
+      </Box>
+      <Section>
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/NtosRoboControl.js:28]: Change to <Box mt={1}>

**Comment by stylemistake** [comment]
image

**Pull request merged**